examples/c/ex_col_store.c

```diff
@@ -0,0 +1,582 @@
+        case 3:
+            strcpy(w.country, "NZ");
+            break;
+        case 4:
+            strcpy(w.country, "IND");
+            break;
+        case 5:
+            strcpy(w.country, "CHI");
+            break;
+        case 6:
+            strcpy(w.country, "RUS");
+            break;
+        default:
+            assert(false);
+        }
+
+        w_array[i] = w;
+    }
+}
+
+static int
+find_min_temp(WT_SESSION *session, uint16_t start_time, uint16_t end_time, int *min_temp)
+{
+    WT_CURSOR *join_cursor, *start_time_cursor, *end_time_cursor;
+    int exact;
+    int ret;
+    uint64_t recno;
+    uint16_t hour;
+    uint8_t temp;
+
+    /* Open cursors needed by the join. */
+    error_check(
+      session->open_cursor(session, "join:table:weather(hour,temp)", NULL, NULL, &join_cursor));
+    error_check(
+      session->open_cursor(session, "index:weather:hour", NULL, NULL, &start_time_cursor));
+    error_check(session->open_cursor(session, "index:weather:hour", NULL, NULL, &end_time_cursor));
+
+    /*
+     * Select values WHERE (hour >= start AND hour <= end). Find the starting record closest to
+     * desired start time.
+     */
+    start_time_cursor->set_key(start_time_cursor, start_time);
+    error_check(start_time_cursor->search_near(start_time_cursor, &exact));
+    ret = 0;
+    if (exact == -1)
+        ret = start_time_cursor->next(start_time_cursor);
+
+    if (ret == 0)
+        error_check(session->join(session, join_cursor, start_time_cursor, "compare=ge"));
+    else
+        return ret;
+
+    /* Find the ending record closest to desired end time. */
+    end_time_cursor->set_key(end_time_cursor, end_time);
+    error_check(end_time_cursor->search_near(end_time_cursor, &exact));
+    if (exact == 1)
+        ret = end_time_cursor->prev(end_time_cursor);
+
+    if (ret == 0)
+        error_check(session->join(session, join_cursor, end_time_cursor, "compare=le"));
+    else
+        return ret;
+
+    /* Initialize minimum temperature to temperature of the first record. */
+    ret = join_cursor->next(join_cursor);
+    if (ret != 0)
+        return ret;
+
+    error_check(join_cursor->get_key(join_cursor, &recno));
+    error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+    *min_temp = temp;
+
+    /* Iterating through found records between start and end time to find the minimum temperature.
+     */
+    while ((ret = join_cursor->next(join_cursor)) == 0) {
+        error_check(join_cursor->get_key(join_cursor, &recno));
+        error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+
+        if (temp < *min_temp)
+            *min_temp = temp;
+
+        /* For debugging purposes */
+        /*
+            printf("ID %" PRIu64, recno);
+            printf(": hour %" PRIu16 " temp: %" PRIu8 "\n", hour, temp);
+         */
+    }
+
+    return (EXIT_SUCCESS);
+}
+
+static int
+find_max_temp(WT_SESSION *session, uint16_t start_time, uint16_t end_time, int *max_temp)
```

Comment:
Most of the min and max temp calculation code is a duplicate code. How about writing a new internal function that performs both min and max calculations and returns it to the caller function to decide which temperature they want to print?

Reply:
Thanks for pointing this out. I will merge these into one function that finds both the min and max temp and writes the results to two different variables.